pkg/model/autoscalinggroup.go

```diff
@@ -0,0 +1,173 @@
+package model
+
+import (
+	"fmt"
+	"k8s.io/kops/pkg/apis/kops"
+	"k8s.io/kops/pkg/model/resources"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/awstasks"
+	"k8s.io/kops/upup/pkg/fi/nodeup"
+	"text/template"
+)
+
+const (
+	DefaultVolumeSize = 20
+	DefaultVolumeType = "gp2"
+)
+
+// AutoscalingGroupModelBuilder configures AutoscalingGroup objects
+type AutoscalingGroupModelBuilder struct {
+	*KopsModelContext
+
+	NodeUpSource     string
+	NodeUpSourceHash string
+
+	NodeUpConfigBuilder func(ig *kops.InstanceGroup) (*nodeup.NodeUpConfig, error)
+}
+
+var _ fi.ModelBuilder = &AutoscalingGroupModelBuilder{}
+
+func (b *AutoscalingGroupModelBuilder) Build(c *fi.ModelBuilderContext) error {
+	for _, ig := range b.InstanceGroups {
```

Comment:
Not if they are already formatted.  In general if we're calling into our own code, we try to return a formatted error.  But then if we have nothing to add, we don't.